### PR TITLE
Update ddl2cpp and add tests

### PR DESCRIPTION
**Pull request by Yaraslaut**
Adapt ddl2cpp to use new DataMapper and add tests

### CODE REVIEW

File: src/benchmark/tables.hpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#include <Lightweight/DataMapper/DataMapper.hpp>
```
Comment (christianparpart) [inline src/benchmark/tables.hpp:1]: `#pragma once` missing, for it being a header file

could we also somehow make it add a SPDX license tag, defaulting to Apache-2.0, but letting ddl2cpp's CLI arguments parametrize it?

File: src/benchmark/tables.hpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#include <Lightweight/SqlConnection.hpp>
+#include <Lightweight/SqlDataBinder.hpp>
+#include <Lightweight/SqlQuery.hpp>
+#include <Lightweight/SqlQueryFormatter.hpp>
+#include <Lightweight/SqlScopedTraceLogger.hpp>
+#include <Lightweight/SqlStatement.hpp>
+#include <Lightweight/SqlTransaction.hpp>
+struct lists;
```
Comment (christianparpart) [inline src/benchmark/tables.hpp:9]:
```suggestion

struct lists;
```

just a readability thought

File: src/tools/ddl2cpp.cpp [hunk 3]
```diff
@@ -47,7 +48,7 @@ std::string MakeType(SqlSchema::Column const& column)
             else
                 return std::format("std::string", column.size);
         case ColumnType::TEXT:
-            return std::format("SqlText");
+            return std::format("SqlString<30>");
         case ColumnType::BOOLEAN:
```
Comment (christianparpart) [inline src/tools/ddl2cpp.cpp:52]: LNB: Due to the lag of information detailness of `enum class ColumnType`, it'll soonish be replaced by `SqlColumnTypeDeclaration` (`variant` over simple structs), which makes it easy to encode text length along with it.

File: src/tools/ddl2cpp.cpp [hunk 4]
```diff
@@ -160,33 +170,27 @@ class CxxModelPrinter
...
+            }
             if (column.isForeignKey)
                 continue;
-            m_definitions << std::format("    Model::Field<{}, {}, \"{}\"{}> {};\n",
-                                         type,
-                                         columnPosition,
-                                         column.name,
-                                         column.isNullable ? ", Nullable" : "",
-                                         column.name);
+            m_definitions << std::format("    Field<{}> {};\n", type, column.name);
```
Comment (christianparpart) [inline src/tools/ddl2cpp.cpp:179]: nullable fields are `std::optional<T>` rather than `T` alone. Is this covered?

File: src/tools/tests/1.sq [hunk 5]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE "Person" (
```
Comment (christianparpart) [inline src/tools/tests/1.sq:1]: why are those file names ending with `.sq` and not `.sql`?

Comment (Yaraslaut) [reply src/tools/tests/1.sq:1]: Changed

File: src/tools/ddl2cpp.cpp (hunk 4 above)
Comment (Yaraslaut) [reply src/tools/ddl2cpp.cpp:179]: Fixed

File: src/benchmark/tables.hpp (hunk 1 above)
Comment (Yaraslaut) [reply src/benchmark/tables.hpp:3]: I added now Apache-2.0 header, can extend it in the future PRs

File: src/tools/ddl2cpp.cpp (hunk 3 above)
Comment (Yaraslaut) [reply src/tools/ddl2cpp.cpp:59]: Returned SqlText for now

**Review by christianparpart — APPROVED** [review]
Thank you very much

**Pull request merged**